pkg/logutils/log.go

```diff
@@ -0,0 +1,190 @@
+	}
+
+	if addFuncCallerSuffix {
+		encoderConfig.EncodeCaller = zapcore.ShortCallerEncoder
+	} else {
+		encoderConfig.EncodeCaller = nil
+	}
+
+	if addLogLevelPrefix {
+		encoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	} else {
+		encoderConfig.EncodeLevel = nil
+	}
+
+	logger := zap.New(zapcore.NewCore(
+		convertToZapFormatEncoder(format, encoderConfig),
+		ws,
+		zap.NewAtomicLevelAt(ConvertToZapLevel(logLevel)),
+	), zap.AddCaller())
+	return logger, nil
+}
+
+// NewLogger create  Logger instance with default configuration for 'stdout' usage, it's JsonLogFormat.
+func NewLogger() (err error) {
+	l, err := NewLoggerWithOption(JsonLogFormat, OUTPUT_STDOUT, nil, true, true, true, LogInfo)
+	if nil != err {
+		return
+	}
+	Logger = l
+	return
+}
+
+// NewLoggerStderr create LoggerStderr instance for 'stderr' usage, it's ConsoleLogFormat.
+// It wouldn't provide 'time prefix', 'function caller suffix' and 'log level prefix' in output.
+func NewLoggerStderr() (err error) {
+	l, err := NewLoggerWithOption(ConsoleLogFormat, OUTPUT_STDERR, nil, false, false, false, LogInfo)
+	if nil != err {
+		panic("bad..." + err.Error())
+	}
+	LoggerStderr = l
+	return
+}
+
+// NewLoggerFile sets LoggerFile configuration for 'file output' usage.
+func NewLoggerFile(filePath string) (err error) {
+	ferr := os.MkdirAll(filepath.Dir(filePath), 0755)
+	if ferr != nil {
+		fmt.Fprintf(os.Stderr, "Failed to create path for CNI log file: %v", filepath.Dir(filePath))
+	}
+
+	// MaxSize    = 100 // MB
+	// MaxAge     = 30 // days (no limit)
+	// MaxBackups = 10 // no limit
+	// LocalTime  = false // use computers local time, UTC by default
+	// Compress   = false // compress the rotated log in gzip format
+	fileLoggerConf := FileOutputOption{
+		Filename:   filePath,
+		MaxSize:    100,
+		MaxAge:     30,
+		MaxBackups: 10,
+	}
+	l, err := NewLoggerWithOption(JsonLogFormat, OUTPUT_FILE, &fileLoggerConf, true, true, true, LogInfo)
+	if nil != err {
+		return
+	}
+	LoggerFile = l
+	return
+}
+
+// ConvertToZapLevel converts log level string to zapcore.Level.
+// if error is not nil it will return the info log level
+func ConvertToZapLevel(level LogLevel) zapcore.Level {
+	var lvl zapcore.Level
+	if err := lvl.Set(string(level)); nil != err {
+		fmt.Fprintf(os.Stderr, "error setting log level %q: %s, and it will set as Info level\n", level, err)
+		return zapcore.InfoLevel
+	}
+	return lvl
+}
+
+// convertToZapFormatEncoder converts and validated log format string.
+func convertToZapFormatEncoder(format LogFormat, encoderConfig zapcore.EncoderConfig) zapcore.Encoder {
+	switch format {
+	case ConsoleLogFormat, "":
+		return zapcore.NewConsoleEncoder(encoderConfig)
+	case JsonLogFormat:
+		return zapcore.NewJSONEncoder(encoderConfig)
+	default:
+		fmt.Fprintf(os.Stderr, "unknown log format: %s, supported values json, console and it will be set as %s\n",
+			format, JsonLogFormat)
+		return zapcore.NewConsoleEncoder(encoderConfig)
+	}
```

Comment:
it is make sense to write a init() for other pkg to log
init(){
   InitStdLogger()
   InitStdOutLogger()
   .....
}
and the main func could recall InitStdLogger()